### PR TITLE
Http composed server

### PR DESCRIPTION
**Pull request by ceolin**
Add support to generator

### CODE REVIEW

File: src/modules/flow-metatype/http-composed-server/http-composed-server-code.h [hunk 1]
```diff
@@ -0,0 +1,365 @@
...
+    "    return r;\n" \
+    "}\n" \
+    "static struct http_server *\n" \
+    "server_ref(int32_t port)\n" \
+    "{\n" \
+    "    struct http_server *idata, *sdata = NULL;\n" \
+    "    uint16_t i;\n" \
+    "    if ((port > UINT16_MAX) || port < 0) {\n" \
+    "        SOL_WRN(\"Invalid server port (%\" PRId32 \"). It must be in range \"\n" \
+    "            \"0 - (%" PRId32 "). Using default port  (%\" PRId32 \").\",\n" \
```
Comment (edersondisouza) [inline src/modules/flow-metatype/http-composed-server/http-composed-server-code.h]: It appears that you missed to escape a couple of `"` here.

File: src/modules/flow-metatype/http-composed-server/http-composed-server.c [hunk 2]
```diff
@@ -1050,58 +1086,246 @@ http_composed_server_ports_description(const struct sol_flow_metatype_context *c
...
+        r = sol_buffer_append_printf(out,
+            "    if (!http_composed_server_%.*s_%s_port.base.packet_type) {\n",
+            SOL_STR_SLICE_PRINT(prefix), port->name);
+        SOL_INT_CHECK(r, < 0, r);
+
+        if (strstartswith(port->type, "composed:")) {
+            struct sol_str_slice types;
+            //Removing the composed: prefix
+            types.data = port->type + 9;
+            types.len = strlen(port->type) - 9;
```
Comment (edersondisouza) [inline src/modules/flow-metatype/http-composed-server/http-composed-server.c]: This is a kinda common pattern - not sure if always with str_slice... maybe having something on `sol_str_slice` to do it?

Comment (cabelitos) [inline src/modules/flow-metatype/http-composed-server/http-composed-server.c]: Use `sol_flow_packet_type_from_string()`

Comment (ceolin) [inline src/modules/flow-metatype/http-composed-server/http-composed-server.c]: It's different, thi code is for the generator, the function that was added returns an object.

**Comment by ceolin** [comment]
merged with the fix.

**Pull request closed without merging**